packages/commands/src/index.ts

```diff
@@ -613,15 +613,16 @@ export class CommandRegistry {
    */
   private _executeKeyBinding(binding: CommandRegistry.IKeyBinding): void {
     let { command, args } = binding;
-    if (!this.hasCommand(command) || !this.isEnabled(command, args)) {
+    let newArgs = { ...args, isKeybinding: true };
```

Comment:
Thanks @andrewfulton9 

To move further in the direction of passing the event information.

Should we set isKeybinding with the `binding.keys` instead of a boolean?

Reply:
Great question. For context we discussed exposing a **subset** of the event itself for menus, for an unrelated reason: https://github.com/jupyterlab/jupyterlab/pull/14173#issuecomment-1465292805

I imagine we could go in the direction of having a `ILuminoEvent` in which is (a) fully JSON-serializable, (b) includes context specific to Lumino applications (higher level concepts as `type: "menu"` or `type: "keybinding"`).

Reply:
@fcollonval, do you mean something like `let newArgs = { ...args, isKeybinding: binding.keys };`? That would then allow for more customization in downstream actions. Am I understanding that correctly?

Reply:
I was personally thinking:

```js
newArgs = { ...args, event: { type: "keybinding", keys: binding.keys } }
```

or if we want to "hide" it:

```js
newArgs = { ...args, _luminoEvent: { type: "keybinding", keys: binding.keys } }
```

or something in between. Thoughts?

Reply:
That seems reasonable to me. Would it always be `type` and `keys` for the `_luminoEvent`? Would you just keep `keys` blank if the event type was `menu`?

Reply:
It looks like `ReadonlyPartialJSONObject` is `readonly [key: string]`, so I don't think I can do that without changing `ReadonlyPartialJsonObject` or updating  `CommandRegistry.execute` to include another args type

Reply:
I do not see a problem here. Can you expand? In general the proposed event would be a union of types following a base interface for shared fields like "type". Such an interface/type w should indeed have read-only fields. It should conform to read-only JSON type.

Reply:
I may be misunderstanding an error. when I try to access `newArgs._luminoEvent?.type` I am getting this error:

```
Property 'type' does not exist on type 'string | number | boolean | ReadonlyPartialJSONObject | ReadonlyPartialJSONArray'.
  Property 'type' does not exist on type 'string'.ts(2339)
```

Any idea why it thinks `_luminoEvent` is a string?

Reply:
You will need to add a cast. It should work since I've if the shower values is ReadonlyPartialJSONObject.